Sources/Http/CalculateHashClient.swift

```diff
@@ -17,34 +17,42 @@ import Foundation
 
 final actor CalculateHashClient {
 
-    static func makeRequest(for request: CalculateHashRequest, accessToken: String, oauth2BaseUrl: String) async throws -> CalculateHashResponse {
+    static func makeRequest(for request: CalculateHashRequest, accessToken: String, oauth2BaseUrl: String) async throws -> Result<DocumentDigests, ClientError> {
 
         let endpoint = "/signatures/calculate_hash"
         let baseUrl = oauth2BaseUrl + endpoint
 
         guard let url = URL(string: baseUrl) else {
-            throw ClientError.invalidRequestURL
+            return .failure(ClientError.invalidRequestURL)
         }
 
         var urlRequest = URLRequest(url: url)
         urlRequest.httpMethod = "POST"
         urlRequest.setValue("application/json", forHTTPHeaderField: "Content-Type")
         urlRequest.setValue("Bearer \(accessToken)", forHTTPHeaderField: "Authorization")
 
-        let jsonData = try JSONEncoder().encode(request)
-        urlRequest.httpBody = jsonData
+        do {
+            let jsonData = try JSONEncoder().encode(request)
+            urlRequest.httpBody = jsonData
+        } catch {
+            return .failure(ClientError.encodingFailed)
+        }
 
         let (data, response) = try await URLSession.shared.data(for: urlRequest)
         
-        guard let httpResponse = response as? HTTPURLResponse, (200...299).contains(httpResponse.statusCode) else {
-            throw ClientError.invalidResponse
+        guard let httpResponse = response as? HTTPURLResponse else {
+            return .failure(ClientError.invalidResponse)
         }
 
-        do {
-            return try JSONDecoder().decode(CalculateHashResponse.self, from: data)
-        } catch {
-            throw CalculateHashError.decodingFailed
+        if (200...299).contains(httpResponse.statusCode) {
+            do {
+                let decodedResponse = try JSONDecoder().decode(DocumentDigests.self, from: data)
+                return .success(decodedResponse)
+            } catch {
+                return .failure(ClientError.clientError(data: data, statusCode: httpResponse.statusCode))
```

Comment:
If client error is only used in requests, maybe we can do a conversion to string here so it's immediately human-readable (and save a client having to do the conversion). This means we might have to update the error enum also.